include/seqan3/io/alignment_file/record.hpp

```diff
@@ -0,0 +1,415 @@
+        return get_impl(field_constant<seqan3::field::flag>{}, static_cast<tuple_base_t const &>(*this));
+    }
+
+    //!\brief The mate pair information with the reference name.
+    decltype(auto) mate_reference_id() &&
+    {
+        return std::get<0>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t &&>(*this)));
+    }
+    //!\copydoc seqan3::sam_record::mate_reference_id
+    decltype(auto) mate_reference_id() const &&
+    {
+        return std::get<0>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t const &&>(*this)));
+    }
+    //!\copydoc seqan3::sam_record::mate_reference_id
+    decltype(auto) mate_reference_id() &
+    {
+        return std::get<0>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t &>(*this)));
+    }
+    //!\copydoc seqan3::sam_record::mate_reference_id
+    decltype(auto) mate_reference_id() const &
+    {
+        return std::get<0>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t const &>(*this)));
+    }
+
+    //!\brief The mate pair information with the position.
+    decltype(auto) mate_position() &&
+    {
+        return std::get<1>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t &&>(*this)));
+    }
+    //!\copydoc seqan3::sam_record::mate_position
+    decltype(auto) mate_position() const &&
+    {
+        return std::get<1>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t const &&>(*this)));
+    }
+    //!\copydoc seqan3::sam_record::mate_position
+    decltype(auto) mate_position() &
+    {
+        return std::get<1>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t &>(*this)));
+    }
+    //!\copydoc seqan3::sam_record::mate_position
+    decltype(auto) mate_position() const &
+    {
+        return std::get<1>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t const &>(*this)));
+    }
+
+    //!\brief The mate pair information with the template length.
+    decltype(auto) template_length() &&
+    {
+        return std::get<2>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t &&>(*this)));
+    }
+    //!\copydoc seqan3::sam_record::template_length
+    decltype(auto) template_length() const &&
+    {
+        return std::get<2>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t const &&>(*this)));
+    }
+    //!\copydoc seqan3::sam_record::template_length
+    decltype(auto) template_length() &
+    {
+        return std::get<2>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t &>(*this)));
+    }
+    //!\copydoc seqan3::sam_record::template_length
+    decltype(auto) template_length() const &
+    {
+        return std::get<2>(get_impl(field_constant<seqan3::field::mate>{}, static_cast<tuple_base_t const &>(*this)));
+    }
+
+    //!\brief The mapping quality of the seqan3::field::seq alignment, usually a Phred-scaled score. 
```

Comment:
```suggestion
    //!\brief The mapping quality of the alignment, usually a Phred-scaled score. 
```

I find this redundant...